packages/nimble-components/src/table-column/base/models/column-validator.ts

```diff
@@ -0,0 +1,55 @@
+import type { ValidityObject } from '../../../table/types';
+import type { ColumnInternals } from './column-internals';
+
+type ObjectFromList<T extends readonly string[], V = string> = {
+    [K in T extends readonly (infer U)[] ? U : never]: V;
+};
+
+/**
+ * Base column validator
+ */
+export class ColumnValidator<ValidityFlagNames extends readonly string[]> {
+    protected configValidity: ObjectFromList<ValidityFlagNames, boolean>;
+
+    public constructor(
+        private readonly columnInternals: ColumnInternals<unknown>,
+        configValidityKeys: ValidityFlagNames
+    ) {
+        type ConfigValidity = typeof this.configValidity;
+        this.configValidity = configValidityKeys.reduce(
+            (r, key): ConfigValidity => {
+                return {
+                    ...r,
+                    [key]: false
+                };
+            },
+            // eslint-disable-next-line @typescript-eslint/prefer-reduce-type-parameter
+            {} as ConfigValidity
+        );
+    }
+
+    public isValid(): boolean {
+        return Object.values(this.configValidity).every(x => !x);
+    }
+
+    public getValidity(): ValidityObject {
+        return {
+            ...this.configValidity
+        };
+    }
+
+    protected setConditionValue(
+        name: ValidityFlagNames extends readonly (infer U)[] ? U : never,
+        isInvalid: boolean
+    ): boolean {
```

Comment:
What is the return type here meant to represent? Currently, the function is returning the negation of the input, which doesn't really seem very useful. Should this really return whether the whole object is valid/invalid.

Perhaps having some docs for these functions would also be helpful.

Reply:
Fair point. I'm making this not return a value, and I'm adding function docs.